### PR TITLE
feat(Styleguide): Simplify indicator story status pills

### PR DESCRIPTION
**Pull request by codecaaron**
## Overview
Inspired by: https://circuit.sumup.com/?path=/docs/typography-subheading--base

Makes `StoryStatus` less complicated:
- No longer requires a hover.
- Stable/Volatile/Deprecated are the tags.

Adds a few missing indicators

### PR Checklist

- [ ] Related to Abstract designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0df9ff1c4bca7133b2f3d4--codecademy-styleguide.netlify.app>https://5f0df9ff1c4bca7133b2f3d4--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0df9ff1c4bca7133b2f3d4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0e0210fa04da0b38f96d57--codecademy-styleguide.netlify.app>https://5f0e0210fa04da0b38f96d57--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0e0210fa04da0b38f96d57>Deploy Logs</a></p>

File: packages/styleguide/Blocks/StoryStatus.tsx [hunk 1]
```diff
@@ -2,54 +2,28 @@ import React from 'react';
...
 import { colors } from '@codecademy/gamut-styles/utils/variables';
 
-const Statuses = {
-  ready: { message: 'Production Ready', color: colors.green[500] },
-  inProgress: { message: 'Use with Caution (WIP)', color: colors.yellow[500] },
-  deprecated: { message: 'Do not use (Deprecated)', color: colors.red[500] },
+const STATUS_COLORS = {
+  stable: colors.green[500],
+  volatile: colors.yellow[500],
+  deprecated: colors.red[500],
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/Blocks/StoryStatus.tsx:8]: These are a little bright for backgrounds behind white text. Can we darken them a bit?

Accessibility issue, for me 😄 

```suggestion
  stable: colors.green[800],
  volatile: colors.yellow[700],
  deprecated: colors.red[700],
```

**Comment by codecaaron** [comment]
> Solid step forward, thank you!
> 
> Just a color request around the colors if you don't mind 🤗

Good call, love it!

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0e090766901d01de47f6d0--codecademy-styleguide.netlify.app>https://5f0e090766901d01de47f6d0--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0e090766901d01de47f6d0>Deploy Logs</a></p>

**Pull request merged**